src/bsp/src/posix/src/BSP.cpp

```diff
@@ -1,6 +1,65 @@
-#include "bsp/BSP.h"
+#include "BSP.h"
+#include "ros/ros.h"
+#include <FreeRTOS.h>
+#include <FreeRTOSConfig.h>
+#include <bsp/BSPFactory.h>
+#include <hive_mind/ExampleMessage.h>
+#include <sstream>
+#include <task.h>
+#include <timers.h>
 
 BSP::BSP() = default;
-BSP::~BSP() = default;
+BSP::~BSP() {
+    if (m_rosNodeHandle != NULL) {
+        delete m_rosNodeHandle;
+    }
+}
 
-void BSP::initChip() {}
+/**
+ * @brief Task that kills FreeRTOS when ROS node is stopped
+ */
+void rosWatcher(void* param) {
+    (void)param;
+    const int loopRate = 100;
+
+    while (ros::ok()) {
+        ros::spinOnce();
+        vTaskDelay(loopRate);
+    }
+
+    vTaskEndScheduler();
+}
+
+void exampleTopicPublish(void* param) {
```

Comment:
I know this is just an example, but maybe make a namespace for our topics to regroup them

Reply:
Made all topics relative to the node name (see `new ros::NodeHandle` in `BSP::initChip()`)